### PR TITLE
port of AssemblyWatcher from NUnit2

### PR DESCRIPTION
**Pull request by drSte**
related to #18

### CODE REVIEW

File: src/TestModel/model/AssemblyWatcher.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
+﻿using System;
```
Comment (CharliePoole) [inline src/TestModel/model/AssemblyWatcher.cs]: Each file should have the standard header. Use date range 20xx-2018, where 20xx is whatever the date is for V2.

File: src/TestModel/model/AssemblyWatcher.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+{
+    public class AssemblyWatcher:IAsemblyWatcher
+    {
+        private FileSystemWatcher[] fileWatchers;
+        private FileInfo[] files;
+
+        protected System.Timers.Timer timer;
+        protected string changedAssemblyPath;
+
+        protected FileInfo GetFileInfo(int index)
```
Comment (CharliePoole) [inline src/TestModel/model/AssemblyWatcher.cs]: These protected items can stay for now, but we should review later after #18 is complete. They seem to be used only for a "mock" we create for testing and we may want to use NSubstitute instead. Leave for now.

File: src/TestModel/model/IAsemblyWatcher.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        /// <summary>
+        /// Initializes the watcher with assemblies to observe for changes.
+        /// </summary>
+        /// <param name="delayInMs">The delay in ms.</param>
+        /// <param name="assemblies">The assemblies.</param>
+#if CLR_2_0 || CLR_4_0
+        void Setup(int delayInMs, System.Collections.Generic.IList<string> assemblies);
+#else
+        void Setup(int delayInMs, System.Collections.IList assemblies);
+#endif
```
Comment (CharliePoole) [inline src/TestModel/model/IAsemblyWatcher.cs]: We don't use these defines in the GUI, so we should just keep the generic and add using statement at top.

File: src/TestModel/model/IAsemblyWatcher.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        /// Initializes the watcher with assemblies to observe for changes.
+        /// </summary>
+        /// <param name="delayInMs">The delay in ms.</param>
+        /// <param name="assemblyFileName">Name of the assembly file.</param>
+        void Setup(int delayInMs, string assemblyFileName);
+
+        /// <summary>
+        /// Releases all resources held by the watcher.
+        /// </summary>
+        void FreeResources();
```
Comment (CharliePoole) [inline src/TestModel/model/IAsemblyWatcher.cs]: It probably makes sense to drop this specially named method and have the interface inherit from `IDisposable`.

File: src/TestModel/tests/AssemblyWatcherTests.cs [hunk 5]
```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.IO;
+using NUnit.Framework;
+
+
+namespace TestCentric.Gui.Model
+{
+    [TestFixture]
+    [Platform(Exclude = "Win95,Win98,WinMe")]
```
Comment (CharliePoole) [inline src/TestModel/tests/AssemblyWatcherTests.cs]: The exclusion of ancient platforms is no longer relevant. :smile:

File: src/TestModel/tests/AssemblyWatcherTests.cs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            FileInfo fileInfo = new FileInfo(fileName);
+            fileInfo.Delete();
+
+            FileInfo temp = new FileInfo(tempFileName);
+            if (temp.Exists) temp.Delete();
+        }
+
+        [Test]
+        // TODO: Exclusion should really only apply to Mono on Windows
+        [Platform(Exclude = "Mono")]
```
Comment (CharliePoole) [inline src/TestModel/tests/AssemblyWatcherTests.cs]: After we merge, I'll test on Linux to see if any of these individual test exclusions can be removed. (I assume you are working on  Windows @drSte)

File: src/TestModel/model/AssemblyWatcher.cs (hunk 1 above)
Comment (drSte) [reply src/TestModel/model/AssemblyWatcher.cs]: I forgot about copyright :)

File: src/TestModel/model/IAsemblyWatcher.cs (hunk 3 above)
Comment (drSte) [reply src/TestModel/model/IAsemblyWatcher.cs]: removed

File: src/TestModel/model/IAsemblyWatcher.cs (hunk 4 above)
Comment (drSte) [reply src/TestModel/model/IAsemblyWatcher.cs]: done

File: src/TestModel/tests/AssemblyWatcherTests.cs (hunk 5 above)
Comment (drSte) [reply src/TestModel/tests/AssemblyWatcherTests.cs]: :) done

File: src/TestModel/tests/AssemblyWatcherTests.cs (hunk 6 above)
Comment (drSte) [reply src/TestModel/tests/AssemblyWatcherTests.cs]: Yes, you are right, I am using VS on Windows. It is also my main use case for the moment ;)

File: src/TestModel/model/AssemblyWatcher.cs [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+// ***********************************************************************
+
+using System;
+using System.Collections;
+using System.IO;
+using System.Timers;
+
+namespace TestCentric.Gui.Model
+{
+    public class AssemblyWatcher:IAsemblyWatcher,IDisposable
```
Comment (CharliePoole) [inline src/TestModel/model/AssemblyWatcher.cs]: Let's put `IDisposable` on `IAssemblyWatcher` instead. That implies any `IAssemblyWatcher` has to also be `IDisposable`. Doing it this way says that __this__ `IAssemblyWatcher` is disposable, but not necessarily others.

File: src/TestModel/model/AssemblyWatcher.cs [hunk 8]
```diff
@@ -0,0 +1,156 @@
+﻿// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole
```
Comment (CharliePoole) [inline src/TestModel/model/AssemblyWatcher.cs]: 2002-2018

File: src/TestModel/model/IAsemblyWatcher.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
+﻿// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole
```
Comment (CharliePoole) [inline src/TestModel/model/IAsemblyWatcher.cs]: 2010-2018

File: src/TestModel/model/IAsemblyWatcher.cs [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System.Collections;
+
+namespace TestCentric.Gui.Model
+{
+    public delegate void AssemblyChangedHandler(string fullPath);
+    public interface IAsemblyWatcher
```
Comment (CharliePoole) [inline src/TestModel/model/IAsemblyWatcher.cs]: IAssemblyWatcher : IDisposable

File: src/TestModel/tests/AssemblyWatcherTests.cs [hunk 11]
```diff
@@ -0,0 +1,128 @@
+﻿using System;
```
Comment (CharliePoole) [inline src/TestModel/tests/AssemblyWatcherTests.cs]: Copyright header, 2011-2018

**Pull request merged**